src/python/pants/bsp/goal.py

```diff
@@ -38,38 +39,32 @@ class BSPGoal(BuiltinGoal):
     def activated(cls, union_membership: UnionMembership) -> bool:
         return False
 
-    @classmethod
-    def register_options(cls, register):
-        super().register_options(register)
-        register(
-            "--server",
-            type=bool,
-            default=False,
-            advanced=True,
-            help=(
-                "Run the Build Server Protocol server. Pants will receive BSP RPC requests via the console. "
-                "This should only ever be invoked via the IDE."
-            ),
-        )
-        register(
-            "--runner-env-vars",
-            type=list,
-            member_type=str,
-            default=["PATH"],
-            help=(
-                "Environment variables to set in the BSP runner script when setting up BSP in a repository. "
-                "Entries are either strings in the form `ENV_VAR=value` to set an explicit value; "
-                f"or just `ENV_VAR` to copy the value from Pants' own environment when the {cls.name} goal was run.\n\n"
-                "This option only takes effect when the BSP runner script is written. If the option changes, you "
-                f"must run `{bin_name()} {cls.name}` again to write a new copy of the BSP runner script.\n\n"
-                "Note: The environment variables passed to the Pants BSP server will be those set for your IDE "
-                "and not your shell. For example, on macOS, the IDE is generally launched by `launchd` after "
-                "clicking on a Dock icon, and not from the shell. Thus, any environment variables set for your "
-                "shell will likely not be seen by the Pants BSP server. At the very least, on macOS consider "
-                "writing an explicit PATH into the BSP runner script via this option."
-            ),
-            advanced=True,
-        )
+    server = BoolOption(
```

Comment:
I already tried this. The new declarative options do not work with `BuiltinGoal`'s.

Reply:
Can you elaborate?

Reply:
https://github.com/pantsbuild/pants/pull/14756#discussion_r824251632

Reply:
I filled the pothole: https://github.com/pantsbuild/pants/pull/14762